internal/agent/agent.go

```diff
@@ -598,8 +599,10 @@ func (a *Agent) startPlugins() {
 	// iterate over and start each plugin
 	for _, plugin := range a.plugins {
 		plugin.LogInfo()
-		func(p Plugin) {
-			go p.Run()
+		go func(p Plugin) {
```

Comment:
Probably we need to add an option to check if instrumentation is enabled and only in that case do extra operations, like start transactions or calculating queue len/cap

Reply:
I think that it needs to take into account but I wouldn't do it in all the cases. For example queues are channels and those structures have already the number of elements so I don't expect that len will cause any performance issues. (In fact we are already logging those and not checking if log is enabled). I guess that we should find a balance.